notification/job.go

```diff
@@ -189,7 +189,21 @@ func ProcessPendingNotifications(parentCtx context.Context) (bool, error) {
 			currentHistory.ResourceID,
 		)
 
-		if err := processPendingNotification(ctx, currentHistory); err != nil {
+		var groupedHistory []models.NotificationSendHistory
```

Comment:
I think this is prone to race conditions - We should probably have a different job to group the notifications by groupByPeriod i.e. for each unique hash, `UPDATE send_history set status = Grouped where id in (select id from history where hash=1 and created_at < now-group by period, order by created_at offset 1)`

and then waitFor = max(waitFor, groupByPeriod + 60s)

We also need to update the indexes to match these queruies